### ISSUE TITLE
Q4: Model card updates

### ISSUE THREAD

**Issue by AkshitaB** [allenai/allennlp#4740]
Prioritized fields:
* Model details
* Metrics
* Evaluation data
* Training data

Fill in the prioritized fields in all model cards:

- [x] coref-spanbert
- [x] generation-bart
- [x] glove-sst
- [x] lm-masked-language-model
- [x] lm-next-token-lm-gpt2
- [x] mc-roberta-commonsenseqa
- [x] mc-roberta-piqa
- [x] mc-roberta-swag
- [x] pair-classification-decomposable-attention-elmo
- [x] pair-classification-esim (Missing/old information!)
- [x] pair-classification-roberta-mnli
- [x] pair-classification-roberta-snli
- [x] rc-bidaf
- [x] rc-bidaf-elmo
- [x] rc-naqanet
- [x] rc-transformer-qa
- [x] roberta-sst
- [x] structured-prediction-biaffine-parser
- [x] structured-prediction-constituency-parser
- [x] structured-prediction-srl
- [x] structured-prediction-srl-bert
- [x] tagging-elmo-crf-tagger
- [x] tagging-fine-grained-crf-tagger
- [x] tagging-fine-grained-transformer-crf-tagger

Additional models in the demo:

- [x] All semparse models
- [x] NMN

Remaining fields

* Intended use
* Factors
* Quantitative analyses (wrt chosen factors).
* Ethical considerations
* Caveats and recommendations

Misc Notes:
* Create a list of datasets that are used, and information about them.
* Should a new field similar to `model_type` be added, which specifically mentions the underlying pretrained model? Eg. if a model is finetuned on BERT, it is susceptible to the same limitations as BERT.

**Comment by schmmd** [comment]
Model card information is stored here: https://github.com/allenai/allennlp-models/tree/master/allennlp_models/modelcards
